### PR TITLE
Fixed #2212 Pressing back when viewing a category of a picture from search or explore goes too far

### LINKED ISSUE #2212: Pressing back when viewing a category of a picture from search or explore goes too far
**Summary:** 

When pressing 'back' when viewing a category from media details of in explore or search activity, it returns you to explore/search results rather than the media page.

This doesn't happen if you are viewing through a category.

Should make sense if you view GIFs below.

**Steps to reproduce:** 

1. Go to explore
2. Click the first picture
3. Click the first category
4. Click back
Expected: view the media the user was just viewing
Actual: returns to explore

GIF:

mediaback

Note if viewing "through" a category this doesn't happen, so might want to look at how that is implemented and reuse it. For example:

expected

### PR DESCRIPTION
**Pull request by Divyanshu2132**
**Description (required)**

Fixes #2212  Pressing back when viewing a category of a picture from search or explore goes too far
Removed extra back pressed calling from onResume() in Search activity

**Tests performed (required)**

Tested Beta debug on Asus Zenfone Max Pro M1 with API level 27

### CODE REVIEW

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java [hunk 1]
```diff
@@ -214,21 +214,6 @@ public void onSearchImageClicked(int index) {
...
-        if (supportFragmentManager.getBackStackEntryCount()==1){
-            //FIXME: Temporary fix for screen rotation inside media details. If we don't call onBackPressed then fragment stack is increasing every time.
-            //FIXME: Similar issue like this https://github.com/commons-app/apps-android-commons/issues/894
-            // This is called on screen rotation when user is inside media details. Ideally it should show Media Details but since we are not saving the state now. We are throwing the user to search screen otherwise the app was crashing.
-            // 
-            onBackPressed();
-        }
-        super.onResume();
-    }
-
```
Comment (Divyanshu2132) [inline app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java]: @domdomegg look here are some fixme and an onBackPressed call which I had to remove for this issue

Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java]: If it all still works and there are no issues pressing back or rotating the device then that's okay.

Comment (Divyanshu2132) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java]: Yes I have checked there are no issues on pressing back or rotating the device

File: app/src/main/java/fr/free/nrw/commons/explore/categories/SearchCategoriesRenderer.java [hunk 2]
```diff
@@ -15,7 +15,8 @@
  * presentation logic of individual category in search is handled here
  **/
 class SearchCategoriesRenderer extends Renderer<String> {
-    @BindView(R.id.textView1) TextView tvCategoryName;
+    @BindView(R.id.textView1)
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/explore/categories/SearchCategoriesRenderer.java]: Please undo unrelated change

File: app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java [hunk 3]
```diff
@@ -45,32 +45,32 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         ButterKnife.bind(this, rootView);
         recentSearches = recentSearchesDao.recentSearches(10);
 
-      if(recentSearches.isEmpty()) {
+        if (recentSearches.isEmpty()) {
```
Comment (domdomegg) [inline app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java]: Please submit unrelated whitespace fixes as a separate PR (feel free to request review from me for this)

File: app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java (hunk 1 above)
Comment (domdomegg) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java]: After testing can confirm this is the case. Still kinda worried, as it looks like it should be there for a reason. @ujjwalagrawal17 I believe you initially added this in 046533050e16e447b2d8bc25c816e1fd53dc9933, do you think you could provide feedback on this? Thanks!

Also can we remove the duplicate of this in `CategoryImagesActivity.java` without breaking something? (Best done in a different PR probably)

Comment (ujjwalagrawal17) [reply app/src/main/java/fr/free/nrw/commons/explore/SearchActivity.java]: @domdomegg Umm, when I added this the app was crashing in case of rotation when we are in explore screen, not inside media details. Here is the PR that I sent https://github.com/commons-app/apps-android-commons/pull/1626. We need to make sure https://github.com/commons-app/apps-android-commons/issues/1579 doesn't come again. If it works fine we can merge.

File: app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java (hunk 3 above)
Comment (Divyanshu2132) [reply app/src/main/java/fr/free/nrw/commons/explore/recentsearches/RecentSearchesFragment.java]: I just summit reformatted code

**Comment by domdomegg** [comment]
This does not fix the issue and as it's been stale for over 3 months am closing. Please reopen/submit another PR if this gets fixed.
